src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -59,9 +59,13 @@ def get_remaining_time_in_minutes(self, current_time=None, log_to_stdout=False):
 
         return remaining_time_in_minutes
 
-    def is_package_install_time_available(self, remaining_time_in_minutes=None):
+    def is_package_install_time_available(self, reboot_manager, remaining_time_in_minutes=None, number_of_packages = 1):
```

Comment:
Don't inject reboot_manager like this. What you are looking for is the reboot_setting which you can get from self.execution_config.reboot_setting without doing anything else.

Reply:
Done.